### DISCUSSION TITLE
Where to start with fragment.bed as the only input file?

### DISCUSSION THREAD

**Discussion by HKL-97** [stuart-lab/signac#1025] [Q&A] [answered]
Hello!

We are following a custimozed protocol of scATAC-seq and got a fragment.bed file which is similar to fragments.tsv.gz generated by 10x cellranger.
Does Signac have a workflow that starts from this file as the only input? I guess the next steps should be peak calling and feature counting, but I am not sure what is the best way to do to accommodate with Signac.

Best

**Comment by timoast** [comment, marked as answer]
You can run a full analysis in Signac starting from a fragment file, however a `.bed` file is not the same thing as a fragment file. See [here](https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/output/fragments) for a description of the format. It should be sorted, bgzip-compressed, and indexed with tabix.

Starting from a fragment file, you can run:
1. `CountFragments()` to count the total fragments per cell barcode and determine what barcodes to retain for downstream steps
2. `CallPeaks()` to find peaks
3. `FeatureMatrix()` to quantify peaks per cell
4. `CreateChromatinAssay()` to create an assay from the count matrix

and continue following vignettes on the Signac website

**Reply by HKL-97** [reply]
Hi Tim,

Thank you so much! That sounds promising to me!

I was trying to use FeatureMatrix() but met such error messages:
`Extracting reads overlapping genomic regions
  |                                                  | 0 % ~calculating  Error in sparseMatrix(i = feature.vec, j = cells.in.regions, x = rep(x = 1,  : 
  NA's in (i,j) are not allowed`

Does that mean there are some cells with no features identified and how could I fix that?
Thanks again!

**Reply by timoast** [reply]
@HKL-97 please open an issue with the full code and output of `sessionInfo()`
